crates/anvil/src/config.rs

```diff
@@ -1261,6 +1276,32 @@ latest block number: {latest_block}"
 
         (db, config)
     }
+
+    /// we only use the gas limit value of the block if it is non-zero and the block gas
+    /// limit is enabled, since there are networks where this is not used and is always
+    /// `0x0` which would inevitably result in `OutOfGas` errors as soon as the evm is about to record gas, See also <https://github.com/foundry-rs/foundry/issues/3247>
+    pub(crate) fn fork_gas_limit(&self, block: &Block<WithOtherFields<Transaction>>) -> u128 {
+        if !self.disable_block_gas_limit {
+            if let Some(gas_limit) = self.gas_limit {
+                return gas_limit;
+            } else if block.header.gas_limit > 0 {
+                return block.header.gas_limit;
+            }
+        }
+
+        u64::MAX as u128
+    }
+
+    /// Returns the gas limit to use for the fork
+    ///
+    /// Checks the config for the `disable_block_gas_limit` flag
+    pub(crate) fn gas_limit(&self) -> u128 {
```

Comment:
this isn't used for fork, right?

this should also include the default in the docs